### PR TITLE
8287901: Loom: Failures with -XX:+VerifyStack

### PR DESCRIPTION
**Pull request by pron**
Please review the following change.

Debugging information added to `frame::describe` (used by `JavaThread::print_frame_layout`) was inaccurate, but proved fatal with `VerifyStack`, which uses the same information for verification.

The location of a caller frame's return address and spilled fp, which are located in the callee, were recorded by the caller frame. In the case of the special frame `Continuation.enterSpecial`, those locations were recorded incorrectly. The fix is to record the information while describing the callee frame, rather than the caller.

Passes tier1

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287901](https://bugs.openjdk.org/browse/JDK-8287901): Loom: Failures with -XX:+VerifyStack

### Reviewers
 * [Patricio Chilano Mateo](https://openjdk.java.net/census#pchilanomate) (@pchilano - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/9066/head:pull/9066` \
`$ git checkout pull/9066`

Update a local copy of the PR: \
`$ git checkout pull/9066` \
`$ git pull https://git.openjdk.java.net/jdk pull/9066/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9066`

View PR using the GUI difftool: \
`$ git pr show -t 9066`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/9066.diff">https://git.openjdk.java.net/jdk/pull/9066.diff</a>

</details>

### CODE REVIEW

**Review by pchilano — APPROVED** [review]
Looks good to me.

File: test/jdk/jdk/internal/vm/Continuation/Basic.java [hunk 1]
```diff
@@ -37,7 +37,18 @@
...
+
+/**
+* @test
+* @requires vm.continuations
+* @requires vm.debug
+* @modules java.base/jdk.internal.vm
+* @build java.base/java.lang.StackWalkerHelper
 *
+* @run testng/othervm --enable-preview -XX:+VerifyStack -Xint Basic
+* @run testng/othervm --enable-preview -XX:+VerifyStack -Xcomp -XX:TieredStopAtLevel=3 -XX:CompileOnly=jdk/internal/vm/Continuation,Basic Basic
```
Comment (coleenp) [inline test/jdk/jdk/internal/vm/Continuation/Basic.java:50]: Why 3?  Isn't C1 only equivalent to -XX:TieredStopAtLevel=1 ?

**Review by coleenp — APPROVED** [review]
I have one question about the test.

File: test/jdk/jdk/internal/vm/Continuation/Basic.java (hunk 1 above)
Comment (pron) [reply test/jdk/jdk/internal/vm/Continuation/Basic.java:50]: It's C1 with profiling. Shouldn't matter here.

**Pull request closed without merging**